### PR TITLE
Add Chatbot for User Assistance and Transaction Support on DecenTrade

### LINKED ISSUE #94: [Feature Request]: Add Chatbot for User Assistance and Transaction Support on DecenTrade
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

We would like to add a chatbot to DecenTrade, a decentralized digital marketplace, to enhance user experience and provide real-time assistance. The chatbot will be integrated into the existing platform to help users with queries related to digital asset transactions, account management, and general support.

### Use Case

the chatbot will be beneficial in the following scenario:

User Onboarding: New users often need guidance in navigating decentralized platforms. The chatbot can assist them in understanding how to connect their Ethereum wallets, browse digital assets, and make secure transactions.
Transaction Queries: Users can ask about the status of their transactions, details of their purchases, and potential errors during smart contract interactions.
General FAQs: Users can ask questions about the platform, its features, and how to resolve common issues without needing to contact support

### Benefits

Improved User Experience: The chatbot will offer real-time assistance, reducing user frustration and helping them complete tasks faster.
Reduced Support Load: Common queries can be handled automatically by the bot, reducing the need for human intervention and freeing up resources for more complex issues.
Increased User Engagement: A chatbot creates a more interactive experience, keeping users engaged and helping them complete transactions seamlessly.
Scalability: As the user base of DecenTrade grows, the chatbot will be a scalable solution to handle increased support and transaction-related questions.

### Priority

High

### Record

- [X] I agree to follow this project's Code of Conduct
- [X] I'm a WOB contributor
- [X] I want to work on this issue
- [X] I'm willing to provide further clarification or assistance if needed.

### PR DESCRIPTION
**Pull request by amanver45**
Add Chatbot for User Assistance and Transaction Support on DecenTrade   close #94 

https://github.com/user-attachments/assets/ec16e5cc-0536-4cc3-8c4c-e2924b0d931f

### CODE REVIEW

**Comment by amanver45** [comment]
@4darsh-Dev pls merge this pr and guve level 3

**Review by 4darsh-Dev — CHANGES_REQUESTED** [review]
nice work @amanver45 , 
## Few changes are required, 
see preview : https://decentrade-git-fork-amanver45-main-adarsh-mauryas-projects.vercel.app/
1. Ensure the position is fixed or absolute, so chat UI is visible on home page, 
2. Chat icon should be on pages, u can adjust this in app.jsx

**Comment by amanver45** [comment]
@4darsh-Dev sorry sir pls sir can you merge  this because my mid sem start from Tomorrow so i don't have time

**Comment by amanver45** [comment]
@4darsh-Dev  I can fix after my exam is over

File: frontend/src/App.jsx [hunk 1]
```diff
@@ -1,37 +1,33 @@
...
-import NFTMarketplace from './components/NFTMarketplace'
-import ExplorePage from './pages/ExplorePage'
-import AboutPage from './pages/AboutPage'
-import CreatorsPage from './pages/CreatorsPage'
-import FAQPage from './pages/FAQPage'
-import Navbar from './components/Navbar'
-import CreateNFT from './pages/CreateNFTPage'
-import NotFoundPage from './components/NotFoundPage'
-import Footer from './components/Footer'
-import CustomCursor from './components/CustomCursor'
```
Comment (4darsh-Dev) [inline frontend/src/App.jsx:13]: you've not added custom cursor component,

**Review by 4darsh-Dev — COMMENTED** [review]
@amanver45 , 
your changes do not reflect new cursor component
--

File: frontend/src/App.jsx (hunk 1 above)
Comment (amanver45) [reply frontend/src/App.jsx:13]: app.jsx is updated pls check

**Comment by amanver45** [comment]
@4darsh-Dev  cursor component is added pls check

**Comment by amanver45** [comment]
@4darsh-Dev  pls check

**Review by 4darsh-Dev — CHANGES_REQUESTED** [review]
you also have to use the component 
not only importing will make it effective 
Screenshot 2024-10-24 175919

**Comment by 4darsh-Dev** [comment]
u can see deployment link : https://decentrade-ehma9k3bq-adarsh-mauryas-projects.vercel.app/

**Comment by amanver45** [comment]
@4darsh-Dev custmor cursor added pls check

**Comment by amanver45** [comment]
@4darsh-Dev check this pr

**Comment by amanver45** [comment]
@4darsh-Dev  pls check

**Review by 4darsh-Dev — APPROVED** [review]
fine 👍🏻

**Comment by 4darsh-Dev** [comment]
Gentle Reminder 🚨🚨 
 GSSoC-Ext 24 is going to complete in 2 Days. '
 Complete your assigned Issues and PR reviews before time to get Points on Leaderboard. '
 It was a great experience working with you all, Don't forget to ⭐ star the Repo (only 12 with 62 forks 💔) . 
 Thanks💗 for your valuable contributions!

**Pull request merged**